From 397ac80850bf8d010fae23dd215db0ee2c677306 Mon Sep 17 00:00:00 2001
From: Luca Boccassi <bluca@debian.org>
Date: Thu, 7 May 2020 00:19:40 +0100
Subject: [PATCH] Problem: ZMTP v1 static allocator is needlessly resized

Solution: don't do it, resizing the shared allocator makes sense
as it can take the message buff for zero copy, but the static allocator
is fixed
---
 src/decoder_allocators.hpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/decoder_allocators.hpp b/src/decoder_allocators.hpp
index f6b967be24..2620f1b8ec 100644
--- a/src/decoder_allocators.hpp
+++ b/src/decoder_allocators.hpp
@@ -58,7 +58,8 @@ class c_single_allocator
 
     std::size_t size () const { return _buf_size; }
 
-    void resize (std::size_t new_size_) { _buf_size = new_size_; }
+    //  This buffer is fixed, size must not be changed
+    void resize (std::size_t new_size_) { LIBZMQ_UNUSED (new_size_); }
 
   private:
     std::size_t _buf_size;
